src/main/java/org/bonitasoft/connectors/ws/SecureWSConnector.java

```diff
@@ -363,31 +363,40 @@ private void printRequestAndResponse(Source sourceResponse, boolean buildRespons
         }
     }
 
-    private Node getEnveloppeBodyContent(final Document enveloppe) {
-        final Node enveloppeNode = enveloppe.getFirstChild();
-        final NodeList children = enveloppeNode.getChildNodes();
-        Node enveloppeBody = null;
+    private Node getEnvelopeBodyContent(final Document envelope) {
+        final Node envelopeNode = envelope.getFirstChild();
+        final NodeList children = envelopeNode.getChildNodes();
+        Node envelopeBody = null;
         for (int i = 0; i < children.getLength(); i++) {
             final Node child = children.item(i);
             if (child instanceof Element) {
                 final Element element = (Element) child;
                 if ("Body".equalsIgnoreCase(element.getLocalName())) {
-                    enveloppeBody = child;
+                    envelopeBody = child;
                     break;
                 }
             }
         }
-        if (enveloppeBody == null) {
-            return enveloppeNode;
+        if (envelopeBody == null) {
+            return envelopeNode;
+        }
+
+        final NodeList bodyChildren = envelopeBody.getChildNodes();
+        for (int i = 0; i < bodyChildren.getLength(); i++) {
+            final Node child = bodyChildren.item(i);
+            if (child instanceof Element) {
+                return child;
+            }
         }
 
-        return enveloppeBody.getFirstChild();
+        return envelopeNode;
     }
 
-    private Transformer getTransformer() throws TransformerConfigurationException, TransformerFactoryConfigurationError {
+    private Transformer getTransformer() throws TransformerConfigurationException {
         if (transformer == null) {
             transformer = TransformerFactory.newInstance().newTransformer();
             transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
```

Comment:
can you explain what it does ?

Reply:
I suppose this is to make the formatting use 2 spaces when indenting the output

Reply:
Actually, when using indent = yes with no indent-amout, it just formats the XML with carriage returns, but zero indentation.